scripts/promote.py

```diff
@@ -105,42 +104,34 @@ def check_requirements():
 
 def make_release_notes(src, dst) -> str:
     """
-    produce release notes by retrieving the different commits from src to dst.
+    Produce release notes by retrieving the different commits from src to dst.
     :param src: the source branch
     :param dst: the destination branch
     :return:
     """
+    result = _subprocess(['git', 'log', '--pretty=format:"%s"', f"origin/{src}...origin/{dst}"])
+    commits = "\n".join([f"- {i[1:-1]}" for i in result.split("\n")])
+
     if args.release_notes:
-        subprocess.call([os.environ.get('EDITOR', 'vim'), args.release_notes])
-        with open(args.release_notes, 'r') as file:
-            r_notes = file.read()
-    else:
-        result = _subprocess(['git', 'log', '--pretty=format:"%s"', f"origin/{src}...origin/{dst}"])
-        r_notes = "\n".join([f"- {i[1:-1]}" for i in result.split("\n")])
-        if args.auto:
-            with tempfile.TemporaryDirectory() as temp_path:
-                temp_file = f"{temp_path}/release_notes.txt"
-                with open(temp_file, 'w') as file:
-                    file.write(r_notes)
-                    subprocess.call([os.environ.get('EDITOR', 'vim'), temp_file])
-                with open(temp_file, 'r') as file:
-                    r_notes = file.read()
-    return r_notes
+        with open(args.release_notes, 'w') as f:
+            f.write(commits)
+
+    return commits
 
 
 def commit(src, dst):
     print(_subprocess(['git', 'remote', 'set-url', 'origin',
                        f'https://{token}@github.com/HumanCellAtlas/fusillade.git']))
     print(_subprocess(['git', '-c', 'advice.detachedHead=false', 'checkout', f'origin/{src}']))
-    print(_subprocess(['git', 'checkout', '-B', dst]))
+    print(_subprocess(['git', 'checkout', dst]))
     print(_subprocess(['git', 'push', '--force', 'origin', dst]))
 
 
 def get_current_version(stage: str = None) -> str:
     "check the latest release from github"
     stage = stage if stage else args.stage
     version_url = 'https://api.github.com/repos/HumancellAtlas/fusillade/releases'
-    releases = s.get(version_url).json()
+    releases = requests.Session().get(version_url).json()
```

Comment:
why do you create a new sessions for each request? You could just use requests.get if you don't want to use a sessions

Reply:
Will change.